bin/models.py

```diff
@@ -41,8 +41,7 @@ def get_by_id(cls, ident):
         code = snippet[b'code'].decode('utf-8')
         views_left = int(snippet[b'views_left'].decode('utf-8'))
         parentid = snippet[b'parentid'].decode('ascii')
-        client_ip = bt.request.environ.get('HTTP_X_FORWARDED_FOR') or bt.request.environ.get('REMOTE_ADDR')
-        if views_left == 0 or client_ip in config.WHITELISTED_VIEWS:
+        if views_left == 0 or bt.request.remote_addr in config.WHITELISTED_VIEWS:
```

Comment:
Pas de gros changement mais je pense que ça serait plus lisible :)

    if views_left == 0:
        pass
    elif bt.request.remote_addr in config.WHITELISTED_VIEWS:
        pass

Reply:
On pourrait vérifier le UserAgent d'abord, ca pourrait éviter de check dans les IP inutilement ?
```diff
+ user_agent = bt.request.environ.get('HTTP_USER_AGENT')
...
- elif bt.request.remote_addr in config.WHITELISTED_VIEWS:
+ elif user_agent in config.WHITELISTED_UA and bt.request.remote_addr in config.WHITELISTED_VIEWS:
```
par exemple.

Reply:
Complètement d'accord, ça permet de ne pas transmettre le contenu de la page si l'user-agent match et juste le head, donc, comme étant plus restrictif, imo ça devrait effectivement être vérifié avant la liste des IP whitelistées

Reply:
Si on fait des restrictions sur base du user-agent, c'est directement dans le controller qu'il faut le faire